crates/holochain_zome_types/src/info.rs

```diff
@@ -40,13 +40,19 @@ pub struct AgentInfo {
     /// Same as the initial pubkey if it has never been changed.
     /// The agent can revoke an old key and replace it with a new one, the latest appears here.
     pub agent_latest_pubkey: AgentPubKey,
+    pub chain_head: (HeaderHash, u32, Timestamp),
 }
 
 impl AgentInfo {
-    pub fn new(agent_initial_pubkey: AgentPubKey, agent_latest_pubkey: AgentPubKey) -> Self {
+    pub fn new(
+        agent_initial_pubkey: AgentPubKey,
+        agent_latest_pubkey: AgentPubKey,
+        chain_head: (HeaderHash, u32, Timestamp),
```

Comment:
I see you added a timestamp to the chain head if we have the actual timestamp that will be committed to the chain db while it's in the scratch space. If it is subject to change, I'm a little concerned about exposing it at this level.

I know we have system validation that ensures each chain entry is later than the prior entry, and I think it runs at the end of all the wasm processing when validating commits just before returning the result. So I'm a little concerned about this timestamp changing. But I guess if any timestamp changes, it changes the header hash which would change all following header hashes, so the hash is no more permanent than this timestamp anyway.

So... yeah... Maybe the timestamp will be useful to devs...